pandas/tests/test_base.py

```diff
@@ -1004,6 +1004,16 @@ def f():
         self.assertFalse(hasattr(t, "b"))
 
 
+class TestNumpyCompat(tm.TestCase):
+
+    # See gh-12238
+    def testSearchSorted(self):
+        mixin = IndexOpsMixin()
+        mixin.values = np.array([1, 2, 3])
```

Comment:
no, see how we do for example ``.value_counts`` where an object is constructed and looped thru. pls follow the same pattern